src/nunit.analyzers.tests/DocumentationTests.cs

```diff
@@ -0,0 +1,335 @@
+                                                                              .OrderBy(x => x.Name)
+                                                                              .Select(t => (DiagnosticAnalyzer)Activator.CreateInstance(t))
+                                                                              .ToArray();
+
+        private static readonly IReadOnlyList<DescriptorInfo> DescriptorInfos = Analyzers
+                                                                                .SelectMany(DescriptorInfo.Create)
+                                                                                .ToArray();
+
+        private static IReadOnlyList<DescriptorInfo> DescriptorsWithDocs => DescriptorInfos.Where(d => d.DocumentationFile.Exists)
+                                                                                           .ToArray();
+
+        private static DirectoryInfo RepositoryDirectory => SolutionFile.Find("nunit.analyzers.sln").Directory.Parent;
+
+        private static DirectoryInfo DocumentsDirectory => RepositoryDirectory.EnumerateDirectories("documentation", SearchOption.TopDirectoryOnly).Single();
+
+        [TestCaseSource(nameof(DescriptorInfos))]
+        public void MissingDocs(DescriptorInfo descriptorInfo)
+        {
+            if (!descriptorInfo.DocumentationFile.Exists)
+            {
+                var descriptor = descriptorInfo.Descriptor;
+                var id = descriptor.Id;
+                DumpIfDebug(descriptorInfo.Stub);
+                File.WriteAllText(descriptorInfo.DocumentationFile.Name + ".generated", descriptorInfo.Stub);
+                Assert.Fail($"Documentation is missing for {id}");
+            }
+        }
+
+        [TestCaseSource(nameof(DescriptorInfos))]
+        public void UniqueIds(DescriptorInfo descriptorInfo)
+        {
+            Assert.AreEqual(1, DescriptorInfos.Select(x => x.Descriptor)
+                                              .Distinct()
+                                              .Count(d => d.Id == descriptorInfo.Descriptor.Id));
+        }
+
+        [TestCaseSource(nameof(DescriptorsWithDocs))]
+        public void TitleId(DescriptorInfo descriptorInfo)
+        {
+            Assert.AreEqual($"# {descriptorInfo.Descriptor.Id}", descriptorInfo.DocumentationFile.AllLines.First());
+        }
+
+        [TestCaseSource(nameof(DescriptorsWithDocs))]
+        public void Title(DescriptorInfo descriptorInfo)
+        {
+            var expected = $"## {descriptorInfo.Descriptor.Title}";
+            var actual = descriptorInfo.DocumentationFile.AllLines
+                                       .Skip(1)
+                                       .First()
+                                       .Replace("`", string.Empty);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Explicit("No descriptions. Remove this test and update doc stubs if we don't want descriptions.")]
```

Comment:
This test should be removed or `[Explicit]` should be removed depending on if we want description.

Reply:
We should add descriptions and then remove `[Explicit]`.